Source/WebGPU/WebGPU/BindGroupLayout.mm

```diff
@@ -460,6 +460,9 @@ static void reportErrorInCreateBindGroupLayout(NSString* errorMessage, bool isAu
 
 static bool isEqual(const WGPUBufferBindingLayout& entry, const WGPUBufferBindingLayout& otherEntry)
 {
+    if (entry.type > WGPUBufferBindingType_ReadOnlyStorage || otherEntry.type > WGPUBufferBindingType_ReadOnlyStorage)
```

Comment:
instead of using `>` we could check for the following:
```
WGPUBufferBindingType_Float3x2
WGPUBufferBindingType_Float4x3
WGPUBufferBindingType_ArrayLength 
```
which is equivalent, maybe a static_assert would help